server/protocols/protocols-pop3/src/main/java/org/apache/james/pop3server/mailbox/MailboxAdapter.java

```diff
@@ -174,15 +172,11 @@ public void remove(String... uids) throws IOException {
             .map(uid -> MessageUid.of(Long.valueOf(uid)))
             .collect(Guavate.toImmutableList());
 
-        List<MessageRange> ranges = MessageRange.toRanges(uidList);
         try {
             mailboxManager.startProcessingRequest(session);
-            for (MessageRange range : ranges) {
-                manager.setFlags(new Flags(Flags.Flag.DELETED), MessageManager.FlagsUpdateMode.ADD, range, session);
-                manager.expunge(range, session);
-            }
+            manager.delete(uidList, session);
         } catch (MailboxException e) {
-            throw new IOException("Unable to remove messages for ranges " + ranges);
+            throw new IOException("Unable to remove messages", e);

```

Comment:
I think passing the details of the range is better than passing no information to the log.

Reply:
Except there is no range anymore here, just a list of uids?

Reply:
Can we rather pass an Immutable list of uid? This information is important...